packages/toolpad-components/src/Chart.tsx

```diff
@@ -171,8 +171,8 @@ function Chart({ data = [], loading, error, sx }: ChartProps) {
 
   return (
     <Box sx={{ ...sx, position: 'relative', height: '100%', width: '100%' }} aria-busy={loading}>
-      {displayError ? <ErrorOverlay error={displayError} /> : null}
-      {loading && !error ? (
+      {displayError && !loading ? <ErrorOverlay error={displayError} /> : null}
```

Comment:
This is just an adjustment, doesn't address any of the reported issues in this PR.